core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -86,6 +86,20 @@ public boolean isEmpty() {
     return (globalDeletes == null || globalDeletes.length == 0) && sortedDeletesByPartition.isEmpty();
   }
 
+  public List<DeleteFile> referencedDeleteFiles() {
```

Comment:
Would `Iterable<DeleteFile>` work instead? That would make this lazy.

Reply:
Updated.